clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -988,7 +989,8 @@ long getExpirationTimeForTimeout(final long timeoutMs) {
      * then complete the reconciliation by updating the assignment and making the appropriate state
      * transition. Note that if any of the 2 callbacks fails, the reconciliation should fail.
      */
-    private void revokeAndAssign(SortedSet<TopicIdPartition> assignedTopicIdPartitions,
+    private void revokeAndAssign(LocalAssignmentImpl resolvedAssignment,
```

Comment:
btw, I just noticed the `boolean memberHasRejoined = memberEpochOnReconciliationStart != memberEpoch` condition while reading the code again. We have it in two places. I think that this is wrong because the member epoch could effectively change without leaving. @lianetm I recall that we discussed this a while ago. Do we have a jira to address this?

Reply:
It was this same one [KAFKA-16185](https://issues.apache.org/jira/browse/KAFKA-16185) thought for handling the situation around discarding reconciliations (which is what that rejoin check is for). But I had missed it in the review too. I just created [KAFKA-16375](https://issues.apache.org/jira/browse/KAFKA-16375) to handle it in a follow-up PR (probably moving away of all epochs to cover that edge case, and identifying the rejoin simply by the member going through a transition to join).